apps/extension/src/background/approvals/service.ts

```diff
@@ -21,89 +24,131 @@ export class ApprovalsService {
     protected readonly connectedTabsStore: KVStore<TabStore[]>,
     protected readonly keyRingService: KeyRingService,
     protected readonly ledgerService: LedgerService
-  ) { }
+  ) {}
 
   // Deserialize transfer details and prompt user
   async approveTransfer(txMsg: string, type?: AccountType): Promise<void> {
     const txMsgBuffer = Buffer.from(fromBase64(txMsg));
-    const id = uuid();
-    await this.txStore.set(id, txMsg);
+    const msgId = uuid();
+    await this.txStore.set(msgId, txMsg);
 
     // Decode tx details and launch approval screen
     const txDetails = deserialize(txMsgBuffer, TransferMsgValue);
-    const { source, target, token, amount: amountBN } = txDetails;
+    const {
+      source,
+      target,
+      token: tokenAddress,
+      amount: amountBN,
+      tx: { publicKey = "" },
+    } = txDetails;
     const amount = new BigNumber(amountBN.toString());
-    const baseUrl = `${browser.runtime.getURL(
-      "approvals.html"
-    )}#/approve-transfer`;
+    const baseUrl = `${browser.runtime.getURL("approvals.html")}#/approve-tx/${
+      TxType.Transfer
+    }`;
 
     const url = paramsToUrl(baseUrl, {
-      id,
+      msgId,
       source,
       target,
-      token,
+      tokenAddress,
       amount: amount.toString(),
-      type: type as string,
+      accountType: type as string,
```

Comment:
If `type` is undefined, this will be read in ApproveTx as the string `"undefined"`. I'm not sure if that matters though. By the way, what are the cases of transfers where `type` would be undefined here?

Reply:
Thanks! This was an error from an earlier version of that handler I hadn't fixed yet. `type` is required